services/wfm/src/main/java/org/openkilda/wfm/topology/flow/service/CommandFactory.java

```diff
@@ -0,0 +1,199 @@
+/* Copyright 2018 Telstra Open Source
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+
+package org.openkilda.wfm.topology.flow.service;
+
+import org.openkilda.messaging.command.flow.BaseInstallFlow;
+import org.openkilda.messaging.command.flow.InstallEgressFlow;
+import org.openkilda.messaging.command.flow.InstallIngressFlow;
+import org.openkilda.messaging.command.flow.InstallOneSwitchFlow;
+import org.openkilda.messaging.command.flow.InstallTransitFlow;
+import org.openkilda.messaging.command.flow.RemoveFlow;
+import org.openkilda.messaging.command.switches.DeleteRulesCriteria;
+import org.openkilda.messaging.info.event.PathNode;
+import org.openkilda.messaging.model.Flow;
+import org.openkilda.messaging.model.SwitchId;
+import org.openkilda.messaging.payload.flow.OutputVlanType;
+import org.openkilda.model.FlowSegment;
+import org.openkilda.persistence.PersistenceManager;
+import org.openkilda.persistence.repositories.FlowSegmentRepository;
+import org.openkilda.wfm.share.mappers.FlowMapper;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.UUID;
+
+public class CommandFactory {
+
+    private FlowSegmentRepository flowSegmentRepository;
+
+
+    public CommandFactory(PersistenceManager persistenceManager) {
+        flowSegmentRepository = persistenceManager.getRepositoryFactory().createFlowSegmentRepository();
+    }
+
+    /**
+     * Generates install rules for flow.
+     *
+     * @param flow flow to be installed.
+     * @return list of commands
+     */
+    public List<BaseInstallFlow> getInstallRulesForFlow(Flow flow) {
+        OutputVlanType outputVlanType = getOutputVlanType(flow);
+        if (Objects.equals(flow.getSourceSwitch(), flow.getDestinationSwitch())) {
+            return getOneSwitchRules(flow, outputVlanType);
+        }
+
+        List<BaseInstallFlow> commands = new ArrayList<>();
+
+        List<PathNode> pathNodes = flow.getFlowPath().getPath();
+        if (pathNodes.size() == 0) {
+            return commands;
+        }
+        commands.add(getIngressFlow(flow, outputVlanType));
+
+        for (int i = 1; i < pathNodes.size() - 1; i += 2) {
+            PathNode src = pathNodes.get(i);
+            PathNode dst = pathNodes.get(i + 1);
+            if (!Objects.equals(src.getSwitchId(), dst.getSwitchId())) {
```

Comment:
I think it's OK for now, since the whole change is a huge after all. I'd change this in more atomic way after this pre-h&s activity. But good point.:)

Reply:
ok